### PR TITLE
Moved parallaxgen mismatch warnings to its own class

### PR DESCRIPTION
**Pull request by hakasapl**
Addresses https://github.com/hakasapl/ParallaxGen/issues/50 and https://github.com/hakasapl/ParallaxGen/issues/49

Can't do tests for `ParallaxGenWarnings` yet because I need to improve `BethesdaDirectoryTests` first which currently doesn't have an environment setup for MO2/Vortex mod detection. Also `ParallaxGenConfig` needs tests as well before

### CODE REVIEW

File: ParallaxGenLib/src/ParallaxGenWarnings.cpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  {
+    auto KeyDebug = make_pair(MatchedPath, BaseTex);
+    const lock_guard<mutex> Lock(MismatchWarnDebugTrackerMutex);
+    if (MismatchWarnDebugTracker.find(KeyDebug) != MismatchWarnDebugTracker.end()) {
+      return;
+    }
+
+    MismatchWarnDebugTracker.insert(KeyDebug);
+
+    spdlog::debug(L"[Tex Mismatch] Matched path {} from mod {} does not come from the same diffuse or normal {} from mod {}",
```
Comment (Sebastian-81) [inline ParallaxGenLib/src/ParallaxGenWarnings.cpp:56]: `[Potential Texture Mismatch]` would be better imho

File: ParallaxGenLib/src/ParallaxGenWarnings.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    if (MismatchWarnTracker.find(Key) != MismatchWarnTracker.end()) {
+      return;
+    }
+
+    // add to tracker if not
+    MismatchWarnTracker.insert(Key);
+  }
+
+  // log warning
+  spdlog::warn(L"[Tex Mismatch] Mod \"{}\" assets were used with diffuse or normal from mod \"{}\". Please verify that "
```
Comment (Sebastian-81) [inline ParallaxGenLib/src/ParallaxGenWarnings.cpp:72]: `[Potential Texture Mismatch]` would be better imho

**Review by Sebastian-81 — COMMENTED** [review]
looks good, some minor points, see comments

File: ParallaxGenLib/include/ParallaxGenWarnings.hpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  static std::unordered_set<std::pair<std::wstring, std::wstring>, PairHash>
+      MeshWarnTracker;                    /** Keeps tabs on WARN mesh mismatch messages to avoid duplicates */
+  static std::mutex MeshWarnTrackerMutex; /** Mutex for MeshWarnTracker */
+  static std::unordered_set<std::pair<std::wstring, std::wstring>, PairHash>
+      MeshWarnDebugTracker;                    /** Keeps tabs on DEBUG mesh mismatch messages to avoid duplicates */
+  static std::mutex MeshWarnDebugTrackerMutex; /** Mutex for MeshWarnDebugTracker */
+
+public:
+  /**
```
[the remark this answers is not in the document]
Comment (hakasapl) [reply ParallaxGenLib/include/ParallaxGenWarnings.hpp:41]: Yeah, there will be a point where I want to figure out our linting rules a bit better and we can figure it out then I think

**Review by Sebastian-81 — APPROVED** [review]

**Pull request merged**